### PR TITLE
Changed an expected BIL in x86 mov insn

### PR DESCRIPTION
**Pull request by gitoleg**
Changed an expected BIL due to updated simplification in BAP

### CODE REVIEW

File: mc.tests/x86-mov.exp [hunk 1]
```diff
@@ -55,7 +55,7 @@ set suite(x86-64) {
     {0xb4,0x04} {movb $0x4, %ah} {RAX := high:48\[RAX\].4.low:8\[RAX\]}
 
     {0x66,0xb8,0x04,0x00} {movw $0x4, %ax} {RAX := high:48\[RAX\].4}
-    {0xb8,0x04,0x00,0x00,0x00} {movl $0x4, %eax} {RAX := pad:64\[4\]}
+    {0xb8,0x04,0x00,0x00,0x00} {movl $0x4, %eax} {RAX := 4}
```
Comment (ivg) [inline mc.tests/x86-mov.exp:58]: I'm not sure that this is correct. In fact, the previous code also looks incorrect to me. I'm not sure whether it is a bug in the lifter, or it is something wrong with the transformation, but to my understanding of the `mov` sematic, the right code should look like this:

```pascal
RAX := high:32[RAX].pad:32[4]
```

**Review by ivg — CHANGES_REQUESTED** [review]
The test case enforces incorrect semantics.

I'm not against the change, but there is a problem with the original specification, that we need to investigate.

File: mc.tests/x86-mov.exp (hunk 1 above)
Comment (gitoleg) [reply mc.tests/x86-mov.exp:58]: Let me quote Intel® 64 and IA-32 Architectures Software Developer’s Manual
https://software.intel.com/sites/default/files/managed/39/c5/325462-sdm-vol-1-2abcd-3abcd.pdf
Section 3.4.1.1 General-Purpose Registers in 64-Bit Mode:

• 64-bit operands generate a 64-bit result in the destination general-purpose register.
• 32-bit operands generate a 32-bit result, zero-extended to a 64-bit result in the destination general-purpose
register.
• 8-bit and 16-bit operands generate an 8-bit or 16-bit result. The upper 56 bits or 48 bits (respectively) of the
destination general-purpose register are not modified by the operation. If the result of an 8-bit or 16-bit
operation is intended for 64-bit address calculation, explicitly sign-extend the register to the full 64-bits. 

So it's ok, that in 64 bit mode for 32-bit operands wee see a zero padding for high 32-bits of RAX register and don't see any concatenations like for 8-bit and 16-bit operands

**Review by ivg — APPROVED** [review]

File: mc.tests/x86-mov.exp (hunk 1 above)
Comment (ivg) [reply mc.tests/x86-mov.exp:58]: x86 is the weirdest weirdo

**Pull request merged**